### PR TITLE
Updated app to use new site config structure

### LINKED ISSUE #120: Initial support for updated site config
[Google doc](https://docs.google.com/document/d/1hMP2hY__Ru4NQkDGhiWNPMqoT6YnQkD6Wu8jLCJKYuc/edit)
[Google doc with some proposed updates](https://docs.google.com/document/d/1Ru7chmcGxjMv91R6Zl-hJjcQBjwWF1hbqti82J0UAZk/edit#heading=h.dsnfsmge1a2k) (just for reference; not directly related to this issue)

We're going with an updated config format that changes the way the left nav is generated. This issue covers the work to generate links in the left nav according to the new config

### PR DESCRIPTION
**Pull request by gsidebo**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #120 

#### What's this PR do?
- Adds a management command to override site configs in the local db
- Updates the front and back end to use the new config format (see issue for details), particularly in the site sidebar where content editing links are generated

#### How should this be manually tested?
- Use the new management command to update the local site config for the course site starter
- Modify the override config to add/remove new menu items, then make sure the sidebar looks correct and links to the correct pages with the correct forms

#### Any background context you want to provide?
- I added the YAML file and management command for overriding site configs in the db because the practice of making changes to the course site config, then creating data migrations to adapt the config JSON to match it was not sustainable. We should drop that practice and apply course site config changes in that override YAML file.
- I removed the page/resource constants because this new config is designed to be more generalized/arbitrary. We no longer have a need to limit WebsiteContent records to be either a page or a resource.

#### Screenshots (if appropriate)
ss 2021-03-18 at 15 03 57

### CODE REVIEW

File: static/js/components/SiteSidebar.tsx [hunk 1]
```diff
@@ -1,45 +1,99 @@
...
-  const configItems: ConfigItem[] = website.starter?.config?.collections ?? []
+const collaboratorMenuItem: MenuLink = {
+  name:  "collaborators",
+  label: "Collaborators"
+}
+
+/*
+ * This function adds a "Settings" menu header if it doesn't exist, then adds a link to the Collaborators page
+ */
+const addCollaboratorMenuItem = (
```
Comment (noisecapella) [inline static/js/components/SiteSidebar.tsx:27]: It looks like this function and the `forEach` below do very similar things. Maybe this should be a utility function:

    const addToList = (map: Map<string, Array<NavMenuItem>, key: string, value: NavMenuItem) => {
        const list = map.get(key) ?? []
        map.set(key, [...list, value])
    }

And then below:

    const configMap = new Map<string, Array<ConfigItem>>()
    configItems.forEach((configItem: ConfigItem) => {
      addToList(configMap, configItem.category, configItem)
    })
    if (website.is_admin) {
      addToList(configMap, "Settings", collaboratorMenuItem)
    }

What do you think?
    }

File: static/js/components/forms/SiteAddContentForm.test.tsx [hunk 2]
```diff
@@ -2,39 +2,52 @@ import React from "react"
...
     setFieldValueStub: SinonStub,
-    configItem: ConfigItem,
-    website: Website
+    configItem: ConfigItem
 
   beforeEach(() => {
     sandbox = sinon.createSandbox()
     setFieldValueStub = sinon.stub()
-    website = makeWebsiteDetail()
     // @ts-ignore
```
Comment (noisecapella) [inline static/js/components/forms/SiteAddContentForm.test.tsx:22]: We can remove this line now, it was only there for the `find(...)` function

File: static/js/components/forms/SiteEditForm.test.tsx [hunk 3]
```diff
@@ -5,32 +5,28 @@ import { shallow } from "enzyme"
...
-    content: WebsiteContent,
-    website: Website
+    content: WebsiteContent
 
   beforeEach(() => {
     sandbox = sinon.createSandbox()
     setFieldValueStub = sinon.stub()
-    website = makeWebsiteDetail()
     content = makeWebsiteContentDetail()
     // @ts-ignore
```
Comment (noisecapella) [inline static/js/components/forms/SiteEditForm.test.tsx:28]: We can remove this `@ts-ignore`

**Review by noisecapella — CHANGES_REQUESTED** [review]

**Comment by gsidebo** [comment]
@noisecapella Ready for review again, and your comment reminded me that we should probably rename `SiteEditForm` if we renamed `SiteAddForm`. I added that to the commit as well

**Review by noisecapella — APPROVED** [review]

**Pull request merged**